api/pkg/observability/deployment/identity.go

```diff
@@ -0,0 +1,52 @@
+package deployment
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func enrichIdentityToPod(podSpec corev1.PodSpec, secretName string, containerNames []string) corev1.PodSpec {
+	secretVolume := createVolumeFromSecret(secretName)
+	updatedPodSpec := podSpec.DeepCopy()
+
+	containerExist := false
+	for idx, containerSpec := range updatedPodSpec.Containers {
+		updatedContainerSpec := containerSpec
```

Comment:
just curious, can't we use `containerSpec` directly? why it should be assigned to `updatedContainerSpec`?

Reply:
this is initially to copy the value of the containerSpec but this is necessary due to deepCopy of root object in the above, I'll update the code